src/main/java/com/sourcegraph/webhook/README.md

```diff
@@ -0,0 +1,66 @@
+# Webhooks
+The Sourcegraph Bitbucket plugin provides an internal implementation for webhooks that supports listening to events at a global or project scope.
+
+## Contents
+- [Payload](#payload)
+- [REST API](#rest-api)
+    * [List](#list)
+    * [Create](#create)
+    * [Delete](#delete)
+
+## Payload
+The event payload request is sent when an event is fired. There will be five attempts before the request fails. This will be recorded in the logs.
+
+The request has these headers:
+- `X-Event-Key` - contains the specific event key
+- `X-Hook-ID` - webhook id
+- `X-Hook-Name` - webhook name
+- `X-Hub-Signature` - HMAC SHA256 hash of the payload (request body)
+
+The payload is stored in the request body as a JSON object. They follow the schema specified in [Event Payload](https://confluence.atlassian.com/bitbucketserver0516/event-payload-966061436.html?utm_campaign=in-app-help&utm_medium=in-app-help&utm_source=stash#Eventpayload-repositoryevents) and
+[Bitbucket Entities](https://docs.atlassian.com/bitbucket-server/docs/5.16.0/reference/javascript/JSON.html) for each event.
+
+Example (`pr:merged`):
+<img width="1266" alt="Screen Shot 2019-11-13 at 6 21 57 AM" src="https://user-images.githubusercontent.com/3507526/68772049-fc7a3480-05dd-11ea-9676-707b40fd3daf.png">
+
+
+## REST API
+Interacting with the webhook REST endpoints requires [authentication](https://developer.atlassian.com/server/bitbucket/how-tos/example-basic-authentication/) from a system admin account.
+
+#### List
+```
+$ curl -X GET 'https://${BITBUCKET_SERVER_URL}/rest/sourcegraph-admin/1.0/webhook'
+```
+There will be a 200 response code if successful.
+
+Output: JSON serialized `[]Webhook`  
+Ex: `[{"id":1,"name":"test","scope":"global","identifier":"","events":["pr"],"endpoint":"endpoint","secret":"secret"}]`
+
+#### Create
+```
+curl -X POST 'https://${BITBUCKET_SERVER_URL}/rest/sourcegraph-admin/1.0/webhook' \
+    -H 'Content-Type: application/json' \
+    -d '{"name":"", "scope":"", "identifier":"", "events":["", "", ...], "endpoint":"", "secret":""}'
```

Comment:
I think it would be much easier for reader if `...` are real/usable events:
```suggestion
    -d '{"name":"", "scope":"", "identifier":"", "events":["pr", "pr:opened", "pr:modified"], "endpoint":"", "secret":""}'
```